rancher/dashboard
# Add manual refresh + incremental loading to all list views

Fixes #6890 

- removing check for restriction of types on incremental+manual 
- update masthead to incorporate directly the `ResourceLoadingIndicator` so that it can used by list views with custom mastheads 
- move `catalog.cattle.io.clusterrepo` typeDescription to its rightful component (`TypeDescription`)
- update logic of custom list views to use getters in order to get the "incremental" updates from the loading of data

**Custom list views changed**
- `catalog.cattle.io.app`
- `catalog.cattle.io.clusterrepo`
- `cis.cattle.io.clusterscan`
- `fleet.cattle.io.bundle`
- `fleet.cattle.io.cluster`
- `fleet.cattle.io.clustergroup`
- `fleet.cattle.io.clusterregistrationtoken`
- `fleet.cattle.io.gitrepo`
- `helm.cattle.io.projecthelmchart` -> needs testing
- `logging.banzaicloud.io.clusterflow`
- `logging.banzaicloud.io.flow`
- `management.cattle.io.cluster`
- `management.cattle.io.feature`
- `management.cattle.io.user`
- `monitoring.coreos.com.alertmanagerconfig`
- `namespace`
- `node`
- `persistentvolume`
- `persistentvolumeclaim`
- `provisioning.cattle.io.cluster`
- `service`
- `ui.cattle.io.navlink`
- `workloads` -> feature was already there... Minor tweaks
- `projects/namespaces`

**Not changed**
- `management.cattle.io.setting` - not needed as it's a list view per-se
- `group.principal` -> as per Richard comment, doesn't work for spoofed types
- `harvesterhci.io.management.cluster` -> spoofed type

**TBD**
- Implement changes for packages such as Harvester, Epinio, Elemental

**Testing**
- enable incremental loading and manual refresh features on Global Settings -> Performance (should set a low threshold so that it can be seen in action)
- Go a custom list view
- Make sure incremental loading and manual refresh works properly

FYI @nwmac
